app/forms/concerns/date_field_builder.rb

```diff
@@ -105,7 +105,8 @@ def four_digit_year(year)
     when 4
       year
     when 2
-      year > Time.current.strftime("%y") ? "19#{year}" : "20#{year}"
+      century = @method.eql?(:date_of_birth) && year > Time.current.strftime("%y") ? "19" : "20"
```

Comment:
should we compare as integers for safety

```suggestion
      century = @method.eql?(:date_of_birth) && year.to_i > Time.current.strftime("%y").to_i ? "19" : "20"
```

```
> "6" > Time.current.strftime("%y")
 => true
> "6".to_i > Time.current.strftime("%y").to_i
 => false
 ```
 not that single digits are possible here anyway but....